totalRP3/Modules/TargetFrame/TargetFrame.lua

```diff
@@ -99,15 +114,38 @@ local function onStart()
 				buttonStructure.adapter(buttonStructure, currentTargetID, currentTargetType);
 			end
 
-			if type(buttonStructure.icon) == "table" and buttonStructure.icon.Apply then
-				if buttonStructure.icon:GetFileID() == 516771 then
-					uiButton:SetIconTextureToFile(buttonStructure.icon:GetFileID())
-				else
-					uiButton:SetIconTexture(buttonStructure.icon:GetFileID())
-				end
-			else
+			-- icon > iconAtlas > iconFile so sayeth the Meorawr.
+			if buttonStructure.icon then
 				uiButton:SetIconTexture(buttonStructure.icon);
+			elseif buttonStructure.iconAtlas and C_Texture.GetAtlasInfo(buttonStructure.iconAtlas) then
+				uiButton:SetIconTextureToAtlas(buttonStructure.iconAtlas);
+			elseif buttonStructure.iconFile then
+				uiButton:SetIconTextureToFile(buttonStructure.iconFile)
```

Comment:
```suggestion
				uiButton:SetIconTextureToFile(buttonStructure.iconFile);
```
Lamb sauce required.